android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -671,9 +671,17 @@ constructor(
       .onSuccess { searchResult ->
         searchResult.forEach { currentSearchResult ->
           val includedResources: Map<ResourceType, List<Resource>>? =
-            currentSearchResult.included?.values?.flatten()?.groupBy { it.resourceType }
+            currentSearchResult.included
+              ?.values
+              ?.flatten()
+              ?.distinctBy { it.id }
+              ?.groupBy { it.resourceType }
           val reverseIncludedResources: Map<ResourceType, List<Resource>>? =
-            currentSearchResult.revIncluded?.values?.flatten()?.groupBy { it.resourceType }
+            currentSearchResult.revIncluded
+              ?.values
+              ?.flatten()
+              ?.distinctBy { it.id }
+              ?.groupBy { it.resourceType }
```

Comment:
please add a todo into the code to remove this after https://github.com/google/android-fhir/issues/2512 is merged

Reply:
Done